Libraries/Text/RCTText.m

```diff
@@ -149,6 +165,101 @@ - (void)didMoveToWindow
   }
 }
 
+#pragma mark Sizing
+
+- (CGRect)updateToFitFrame:(CGRect)frame
+{
+  NSLayoutManager *layoutManager = [_textStorage.layoutManagers firstObject];
+  NSTextContainer *textContainer = [layoutManager.textContainers firstObject];
+  [textContainer setLineBreakMode:NSLineBreakByWordWrapping];
+  
+  NSRange glyphRange = NSMakeRange(0, _textStorage.length);
+
+  [self resetDrawnTextStorage];
+
+  CGSize requiredSize = [self calculateSize:_textStorage];
+  NSInteger linesRequired = [self numberOfLinesRequired:layoutManager];
+
+  __block BOOL hitMinimumScale = NO;
+  while ((requiredSize.height > CGRectGetHeight(frame) ||
+         requiredSize.width > CGRectGetWidth(frame) ||
+         (linesRequired > textContainer.maximumNumberOfLines &&
+          textContainer.maximumNumberOfLines != 0))
+         && !hitMinimumScale)
+  {
+    [_textStorage beginEditing];
+    [_textStorage enumerateAttribute:NSFontAttributeName
+                             inRange:glyphRange
+                             options:0
+                          usingBlock:^(UIFont *font, NSRange range, BOOL *stop)
+    {
+      if (font) {
+        UIFont *originalFont = [_originalString attribute:NSFontAttributeName
+                                                  atIndex:range.location
+                                           effectiveRange:&range];
+        UIFont *newFont = [font fontWithSize:font.pointSize - .5];
```

Comment:
I've just updated the other two comments, they both make sense, and I'll push them up shortly but I don't think this one is actually feasible. 

My logic is this, since NSTextStorage is an NSAttributedString subclass there is no limit to the variations on font sizes and faces present in the attributes. For example, an NSAttributedString/NSTextStorage object could contain Helvetica Neue 30pt, Helvetica Bold 120pt, San Francisco 12pt, etc, etc, etc. 

That's why I took the method of decrementing all font faces by the same amount until it fit, or a given face reached minimumFontScale.